internal/cmd/local/local.go

```diff
@@ -24,17 +27,16 @@ func NewCmdLocal(provider k8s.Provider) *cobra.Command {
 				if dnt {
 					telOpts = append(telOpts, telemetry.WithDnt())
 				}
-
-				telClient = telemetry.Get(telOpts...)
+				cfg.TelClient = telemetry.Get(telOpts...)
 			}
-			printProviderDetails(provider)
+			printProviderDetails(cfg.Provider)
```

Comment:
This is where I find it useful to have a singleton Config struct, which can be constructed at the root command (or the root of the command subtree that makes use of those configs), as opposed to threading it through to all of these subcommand constructors.

I also thing its a little odd to take a config struct here, modify it by assigning the `TelClient` and then passing that along. So I would suggest assigning `TelClient` in the caller (parent command)